cdap-common/src/main/java/co/cask/cdap/common/io/SeekableInputStream.java

```diff
@@ -27,25 +27,54 @@
  */
 public abstract class SeekableInputStream extends FilterInputStream implements Seekable {
 
+  private final long length;
+
+  protected SeekableInputStream(InputStream in, long length) {
+    super(in);
+    this.length = length;
+  }
+
+  /**
+   * Returns the seek position from the beginning of the file. If the given
+   * position >= 0, the same value will be returned. Other, the value of (length - pos) will be returned.
```

Comment:
I actually mean (length + pos), not absolute. Anyway, the sign was wrong in the doc.